src/otx/engine/engine.py

```diff
@@ -327,6 +328,61 @@ def export(self, *args, **kwargs) -> None:
         """Export the trained model to OpenVINO Intermediate Representation (IR) or ONNX formats."""
         raise NotImplementedError
 
+    def explain(
+        self,
+        checkpoint: str | Path | None = None,
+        datamodule: EVAL_DATALOADERS | OTXDataModule | None = None,
+        explain_config: ExplainConfig | None = None,
+        **kwargs,
+    ) -> list | None:
+        """Run XAI using the specified model and data.
+
+        Args:
+            checkpoint (str | Path | None, optional): The path to the checkpoint file to load the model from.
+            datamodule (EVAL_DATALOADERS | OTXDataModule | None, optional): The data module to use for predictions.
+            explain_config (ExplainConfig | None, optional): Config used to handle saliency maps.
+            **kwargs: Additional keyword arguments for pl.Trainer configuration.
+
+        Returns:
+            list: Saliency maps.
+
+        Example:
+            >>> engine.explain(
+            ...     datamodule=OTXDataModule(),
+            ...     checkpoint=<checkpoint/path>,
+            ...     explain_config=ExplainConfig(),
+            ... )
+        """
+        from pathlib import Path
+
+        import cv2
+
+        ckpt_path = str(checkpoint) if checkpoint is not None else self.checkpoint
+        if explain_config is None:
+            explain_config = ExplainConfig()
+
+        lit_module = self._build_lightning_module(
+            model=self.model,
+            optimizer=self.optimizer,
+            scheduler=self.scheduler,
+        )
+        if datamodule is None:
+            datamodule = self.datamodule
+        lit_module.meta_info = datamodule.meta_info
+
+        lit_module.model.register_explain_hook()
+
+        self._build_trainer(**kwargs)
+
+        self.trainer.predict(
+            model=lit_module,
+            datamodule=datamodule,
+            ckpt_path=ckpt_path,
+        )
+        saliency_maps = self.trainer.model.model.explain_hook.records
```

Comment:
I hope that we have a PyTorch Lightning callback to flush `explain_hook.records` into the local disk as files periodically. The main memory cannot maintain the raw tensors which have the same size as input images.

Reply:
@vinnamkim As I understand, these returned saliency maps are much smaller than the initial image. These are grayscale images with size [7,7] for classification or approximately [23,31] for detection for each class

Reply:
Even if it's not as memory intensive as I thought, it doesn't seem neat to me to store many things proportional to the dataset size in main memory. Do you mind whether adding todo here to leave it as a backlog?

Reply:
Galina's comment is correct. Which one update - this is not the case for maskrcnn model.
So in the general case it make sense to manage this memory. Will add todo.